src/backend/app.js

```diff
@@ -27,6 +27,86 @@ app.use(cors());
 
 router.use("/meals", mealsRouter);
 
+// For localhost:5001
+app.get("/", (req, res) => {
+  res.send("Hello world");
+});
+
+// 	Respond with all meals in the future (relative to the when datetime)
+app.get("/future-meals", async (req, res) => {
+  try {
+    const futureMeals = await knex("meal")
+      .select()
+      .where("meal_time", ">", new Date());
+    if (futureMeals.length === 0) {
+      res.status(404).send("No meals found");
+    } else {
+      res.status(200).json(futureMeals);
+    }
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+});
+
+// Respond with all meals in the past (relative to the when datetime)
+app.get("/past-meals", async (req, res) => {
+  try {
+    const pastMeals = await knex("meal")
+    .select()
+    .where("meal_time", "<", new Date());
+    if (pastMeals === 0) {
+      res.status(404).send("No meals found");
+    } else {
+      res.status(200).json(pastMeals);
+    }
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Internal Server Error");
+  }
+});
+
+// Respond with all meals sorted by ID
+app.get("/all-meals", async (req, res) => {
+  try {
+    const allMeals = await knex("meal")
+      .select()
+      .orderBy("id");
+    res.status(200).json(allMeals)
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Error");
+  }
+});
+
+// Respond with the first meal (meaning with the minimum id)
+app.get("/first-meal", async (req, res) => {
+  try {
+    const firstMeal = await knex("meal")
+      .select()
+      .orderBy("id")
+      .first();
+    res.status(200).json(firstMeal)
+  } catch (error) {
+    console.error(error);
+    res.status(500).send("Error");
```

Comment:
Nicely written query! This route requests one singular meal, and if that one is not found, the status `404` should be returned with an appropriate message. Same thing applies for `/last-meal`.

Reply:
So it should be: `    if (firstMeal.length === 0){
      res.status(404).send("No meal found");
    }else {
    res.status(200).json(firstMeal);
    }`

Reply:
Yes, that would definitely work!

Reply:
thank you😊